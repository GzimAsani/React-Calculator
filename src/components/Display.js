import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ total, next }) => (
  <div>
    <span>{(next && next.toString()) || (total && total.toString()) || '0'}</span>
  </div>
);

Display.propTypes = {
  total: PropTypes.string,
  next: PropTypes.string,
};

Display.defaultProps = { total: '', next: '' };

export default Display;
