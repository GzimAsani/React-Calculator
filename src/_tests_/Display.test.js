import { create } from 'react-test-renderer';
import Display from '../components/Display';

test('renders correctly', () => {
  const tree = create(<Display />).toJSON();
  expect(tree).toMatchSnapshot();
});
