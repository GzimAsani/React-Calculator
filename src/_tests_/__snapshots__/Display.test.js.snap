// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <span>
    0
  </span>
</div>
`;
